com.unity.render-pipelines.core/Runtime/Volume/VolumeParameter.cs

```diff
@@ -1641,14 +1642,239 @@ internal override void Interp(VolumeParameter from, VolumeParameter to, float t)
+                // only remaining case is that we have a proper index
+                ret = keys[index];
+            }
+            return ret;
+        }
+
+        static private void EvalCurveSegmentAndDeriv(ref float dstValue, ref float dstDeriv, Keyframe lhsKey, Keyframe rhsKey, float currTime)
+        {
+            Assert.IsTrue(lhsKey.time <= rhsKey.time);
+
+            float dx = Mathf.Max(rhsKey.time - lhsKey.time, 0.0001f);
+            float dy = rhsKey.value - lhsKey.value;
+            float length = 1.0f / dx;
+            float lengthSqr = length * length;
+
+            float m1 = lhsKey.outTangent;
+            float m2 = rhsKey.inTangent;
+            float d1 = m1 * dx;
+            float d2 = m2 * dx;
+
+            // Note: The coeffecients are calculated to match what the editor does internally. These coeffeceients expect a
+            // t in the range of [0,dx]. We could change the function to accept a range between [0,1], but then this logic would
+            // be different from internal editor logic which could cause subtle bugs later.
 
-        // TODO: Curve interpolation
+            float c0 = (d1 + d2 - dy - dy) * lengthSqr * length;
+            float c1 = (dy + dy + dy - d1 - d1 - d2) * lengthSqr;
+            float c2 = m1;
+            float c3 = lhsKey.value;
+
+            float t = Mathf.Clamp(currTime - lhsKey.time, 0.0f, dx);
+
+            dstValue = (t * (t * (t * c0 + c1) + c2)) + c3;
+            dstDeriv = (t * (3.0f * t * c0 + 2.0f * c1)) + c2;
+        }
+
+        // lhsIndex and rhsIndex are the indices in the keys array. The lhsIndex/rhsIndex may be -1, in which it creates a synthetic first key
+        // at startTime, or beyond the length of the array, in which case it creates a synthetic key at endTime.
+        static private Keyframe EvalKeyAtTime(Keyframe[] keys, int lhsIndex, int rhsIndex, float startTime, float endTime, float currTime)
+        {
+            Assert.IsTrue(keys.Length >= 1);
+            Assert.IsTrue(startTime <= keys[0].time);
+            Assert.IsTrue(keys[keys.Length - 1].time <= endTime);
+
+            Keyframe lhsKey = FetchKeyFromIndexClamped(keys, lhsIndex, startTime, endTime);
+            Keyframe rhsKey = FetchKeyFromIndexClamped(keys, rhsIndex, startTime, endTime);
+
+            float currValue = 0.0f;
+            float currDeriv = 0.0f;
+            EvalCurveSegmentAndDeriv(ref currValue, ref currDeriv, lhsKey, rhsKey, currTime);
+
+            Keyframe dstKey = new Keyframe(currTime, currValue, currDeriv, currDeriv);
+            return dstKey;
+        }
+
+        public override void Interp(AnimationCurve lhsCurve, AnimationCurve rhsCurve, float t)
+        {
+            if (t <= 0.0f)
+            {
+                m_Value = lhsCurve;
+            }
+            else if (t >= 1.0f)
+            {
+                m_Value = rhsCurve;
+            }
+            else
+            {
+                m_Value = new AnimationCurve();
```

Comment:
Also, this assignment is useless, there is another `new AnimationCurve()` at the end of the function that overwrites it.
But shouldn't there be some sort of cache for these interpolated value, it can add several objects to garbage collection every frame, or maybe it's negligible ?

Reply:
We don't allow GC at runtime in Graphics repository. This will create GC at each interpolation so at each frame if such an animation curve exists in the scene. This is normally detected by one of the automated test but I am not sure he will be able to catch it as I dunno if the scene it use use this new interpolation scheme.

Reply:
Ok. What's the best solution then. Create a pool?

Reply:
There is several point that can be improved, like you can work in usafe and alocate your array on the stack (you create an array of keyframe at some point). But ultimately, if you really want it to be an AnimationCurve, then yes. I don't see any solution beside having a pool. 
Alternatively, you need to define an AnimationCurve like struct that is interpolatable and use it here. 

We are taking a deeper look into this PR in the team. There is improvement on the code side (naming, preventing miss-use, etc). We will check if we see any other good way to handle this interpolation without causing GC. I'll have all the answear Monday :)